src/vscode-dts/vscode.proposed.codeActionRanges.d.ts

```diff
@@ -0,0 +1,16 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	export interface CodeAction {
+		/**
+		 * A range of text that should be highlighted from the Code Action.
+		 *
+		 * Ex: A refactoring action will highlight the range of text that will be changed.
+		 */
+		editRanges?: Range[];
```

Comment:
cc. @mjbvz not sure if this would be the correct type? has some issues since this uses vs codes range, rather than IRange or anything in https://github.com/microsoft/vscode/blob/0bb69da87ca19412a768d5966fcb0e7b36b2dd3f/src/vs/editor/common/core/range.ts#L33C21-L33C21

Reply:
In the APIs you want to make sure you are using `vscode.Range`. Then in the implementation, you can either use `IRange` or  our `Range` type